src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -176,15 +184,29 @@ protected override void XReadActivated(Pipe pipe)
                         // If the subscription is not a duplicate, store it so that it can be
                         // passed to used on next recv call.
                         if (m_options.SocketType == ZmqSocketType.Xpub && (unique || m_verbose))
-                            m_pending.Enqueue(sub.CloneData());
+                        {
+                            m_pending.Enqueue(sub);
+                        }
+                        else
+                        {
+                            sub.Close();
+                        }
+
                     }
                 }
+                else if (m_broadcastEnabled && size > 0 && sub.HasMore && sub[0] == 2)
+                {
+                    m_lastPipe = pipe;
+                    m_lastPipeIsBroadcast = true;
+                    sub.Offset = sub.Offset + 1;
+                    sub.Size = sub.Size - 1;
+                    m_pending.Enqueue(sub);
```

Comment:
Wait, don't change the match logic. It will affect too many areas. The should just remove the prefix before calling the send.

So the user check the prefix byte, if it is 2, remove the prefix and send the message.